swap/src/bob/swap.rs

```diff
@@ -370,7 +370,18 @@ where
                     .await
                     .is_err()
                 {
-                    state4.submit_tx_cancel(bitcoin_wallet.as_ref()).await?;
+                    let result = state4.submit_tx_cancel(bitcoin_wallet.as_ref()).await;
+                    if let Err(error) = result {
+                        let json_rpc_err = error
+                            .downcast_ref::<jsonrpc_client::JsonRpcError>()
+                            .ok_or_else(|| anyhow!("Failed to downcast JsonRpcError"))?;
```

Comment:
```suggestion
                            .context("Failed to broadcast cancel transaction")?;
```
If you are not able to downcast it means that something else went wrong. Maybe Bitcoind was not reachable and hence it is not a `JsonRpcError`.
In this case, we need to have a useful error message.

Reply:
True, that has to be improved! I think it would be better to pass on the original error in this case instead of creating our own `anyhow` error.

Reply:
You were not passing the original error in the first place tho.

Reply:
Now we do :)